org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.java

```diff
@@ -126,6 +126,9 @@ public JvmDeclaredType createType(TypeDeclaration type, String packageName) {
 		case TypeDeclaration.ANNOTATION_TYPE_DECL:
 			jvmType = TypesFactory.eINSTANCE.createJvmAnnotationType();
 			break;
+		case TypeDeclaration.RECORD_DECL:
```

Comment:
its a final. so it will get inlined anyway. afaik

Reply:
Even better ;)